test/upgrade/prober/configuration.go

```diff
@@ -62,22 +64,49 @@ func (p *prober) annotateNamespace() {
 	ensure.NoError(err)
 }
 
+func (p *prober) fetchBrokerUrl() (*apis.URL, error) {
+	namespace := p.config.Namespace
+	p.log.Debugf("Fetching %s broker URL for ns %s", brokerName, namespace)
+	meta := resources.NewMetaResource(brokerName, p.config.Namespace, lib.BrokerTypeMeta)
+	err := duck.WaitForResourceReady(p.client.Dynamic, meta)
+	if err != nil {
+		return nil, err
+	}
+	brokers := p.client.Eventing.EventingV1alpha1().Brokers(namespace)
+	broker, err := brokers.Get(brokerName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	url := broker.Status.Address.URL
+	p.log.Debugf("%s broker URL for ns %s is %v", brokerName, namespace, url)
+	return url, nil
+}
+
 func (p *prober) deployConfigMap() {
 	name := configName
-	p.log.Infof("Deploying config map: %v", name)
+	lib.WaitFor(fmt.Sprintf("configmap be deployed: %v", name), func() error {
+		p.log.Infof("Deploying config map: %v", name)
 
-	configData := p.compileTemplate(configFilename)
-	data := make(map[string]string, 0)
-	data[configFilename] = configData
-	secret := &corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: name,
-		},
-		Data: data,
-	}
-	_, err := p.client.Kube.Kube.CoreV1().ConfigMaps(p.config.Namespace).
-		Create(secret)
-	ensure.NoError(err)
+		brokerUrl, err := p.fetchBrokerUrl()
+		if err != nil {
+			return err
+		}
+		configData := p.compileTemplate(configFilename, brokerUrl)
+		data := make(map[string]string, 0)
+		data[configFilename] = configData
+		watholaConfigMap := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Data: data,
+		}
+		configmaps := p.client.Kube.Kube.CoreV1().ConfigMaps(p.config.Namespace)
```

Comment:
same here with creating the config in a single line, catching the err, etc

Reply:
np